### ISSUE TITLE
Coursera-IBM-Machine-Learning-with-Python-Final-Project

### ISSUE THREAD

**Issue by swatisolanki2406** [skhiearth/Coursera-IBM-Machine-Learning-with-Python-Final-Project#1]
![Screenshot (85)](https://user-images.githubusercontent.com/55667317/82476605-f5336680-9aeb-11ea-9728-1b5a815f9761.png)
I'm facing an issue in Load Test set for evaluation. How can I solve it??

**Comment by thebuffdude** [comment]
I am facing the same issue

**Comment by excalibur768** [comment]
The error is due to one hot encoding not being done for the gender column as it was done on the training data, use the below updated code it should resolve this issue
# convert date time
test_df['due_date'] = pd.to_datetime(test_df['due_date'])
test_df['effective_date'] = pd.to_datetime(test_df['effective_date'])
test_df['dayofweek'] = test_df['effective_date'].dt.dayofweek
# evaulate weekend field
test_df['weekend'] = test_df['dayofweek'].apply(lambda x: 1 if (x>3)  else 0)
# work out education level
test_df['Gender'].replace(to_replace=['male','female'], value=[0,1],inplace=True)
test_feature = test_df[['Principal','terms','age','Gender','weekend']]
test_feature = pd.concat([test_feature,pd.get_dummies(test_df['education'])], axis=1)
test_feature.drop(['Master or Above'], axis = 1,inplace=True)
test_feature.head()
# normalize the test data

test_X = preprocessing.StandardScaler().fit(test_feature).transform(test_feature)
test_X[0:5]
# and target result
test_y = test_df['loan_status'].values
test_y[0:5]
